sympy/core/tests/test_function.py

```diff
@@ -79,6 +80,13 @@ def test_derivative_evaluate():
     assert Derivative(sin(x), x, 0) == sin(x)
 
 
+def test_derivative_reuse_dummy():
+    # this chain rule should introduce two and only two Dummy variables
+    A = f(g(x), h(g(x))).diff(x)
+    dummies = {d for d in preorder_traversal(A) if d.is_Dummy}
+    assert len(dummies) == 2
```

Comment:
Can we just use `.atoms(Dummy)` here?

Reply:
It's been a long time since I looked at this...  That certainly looks simpler.